File: AuthService/src/Application/Interfaces/IRedisService.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Application.Interfaces
{
    public  interface IRedisService
    {
        Task SetStringAsync(string key, string value, TimeSpan? expiry = null);
      
        Task<string?> GetStringAsync(string key);

        Task DeleteAsync(string key);

        Task<bool> ExistsAsync(string key);

        Task<long> IncrementAsync(string key, TimeSpan? expiry = null);

        Task SetObjectAsync<T>(string key, T data, TimeSpan? expiry = null);

        Task<T?> GetObjectAsync<T>(string key);

        Task<string[]> SearchKeysAsync(string pattern);
    }
}

File: ProductService/src/Application/Interfaces/IRedisService.js
class IRedisService {

  async setStringAsync(key, value, expiryInSeconds = null) {
    throw new Error("Method not implemented");
  }

  async getStringAsync(key) {
    throw new Error("Method not implemented");
  }

  async deleteAsync(key) {
    throw new Error("Method not implemented");
  }

  async existsAsync(key) {
    throw new Error("Method not implemented");
  }

  async incrementAsync(key, expiryInSeconds = null) {
    throw new Error("Method not implemented");
  }

 
  async setObjectAsync(key, data, expiryInSeconds = null) {
    throw new Error("Method not implemented");
  }

  async getObjectAsync(key) {
    throw new Error("Method not implemented");
  }

  async searchKeysAsync(pattern) {
    throw new Error("Method not implemented");
  }
}

module.exports = IRedisService;
